hledger-lib/Hledger/Reports/PostingsReport.hs

```diff
@@ -61,15 +65,44 @@ instance HasAmounts PostingsReportItem where
 -- with a tuple.
 type SummaryPosting = (Posting, Period)
 
+-- Possible value expressions taken by the --sort command
+-- Each of these takes a bool, which shows if it has been inverted
+-- (True -> has been inverted, reverse the order)
+data ValueExp
```

Comment:
So far, I've only added support for `date`, `account`, and `amount` as `value-expressions`, although [there are many more](https://manpages.org/ledger) in `ledger`. I felt like this was a reasonable subset to start with, and it's straightforward to add more as needed/requested, either in this PR or down the road.

Reply:
I've seen people ask to sort by amount, but not by account name. Is that (or any other sort field) actually useful ? And multi-parameter sorting, and reversing ? These are cool and might indeed be useful, but I for one am wary of overengineering ahead of real world usefulness.

If not, there's an existing balance --sort-amount flag/sort_amount_ field we could reuse. (And even if so, perhaps we can still adapt and reuse it.)

When sorting by amount, what will happen with the running total/running balance column ?

Reply:
Sure, there's plenty of room for discussion here. I started out pretty much porting the exact behavior of `--sort` from `ledger` (the test cases are direct copies from `ledger`'s output), and I tried to choose what seemed like, at a quick glance, a reasonable subset of sortable fields from the ~60 it supports.

Personally, I do think sorting by account name is useful. For testing while developing, I often ran `hledger register --sort account tag:trip=RecentTrip`, which let me see all of my individual transactions over the trip in `Expenses:Dining`, `Expenses:Travel`, etc. grouped together. This is more granular than just `hledger balance tag:trip=RecentTrip`, and I enjoyed having access to it. However, I'm very open to switching out to any set of fields that are deemed useful by consensus.

The running balance column shows the total from each individual transaction in whatever order they're in.

@igxlin As the original requester in #2180, any thoughts on what fields are most useful?

Reply:
Hi @reesmichael1 , personally, I only use `amount` a lot, as I want to know why I spent that much money from time to time. :see_no_evil: `date`, `amount`, `account` are good enough from my perspective.

Reply:
Fair enough, date/amount/account then. 

Running total just summing items in display order as usual, makes sense. In -H mode, where register shows historical balances after each posting, I'm not sure there's anything we can show that wouldn't be misleading ? Let's see.

Reply:
Thanks everyone! I think this is mostly ready to go now, although I'll flag that I added an error message when someone tries to use both `--sort` and `--historical`. If you think it's better to allow that and live with the misleading results, I'm happy to remove it.